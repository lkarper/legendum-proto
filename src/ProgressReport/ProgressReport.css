.ProgressReport__section {
    display: flex;
    flex-direction: column;
    padding: 5px;
    width: 100%;
}

.ProgressReport__section h3 {
    font-size: 24px;
    margin: 0;
    padding: 5px;
}

.ProgressReport__ol {
    margin: 0;
    padding-left: 20px;
}

.ProgressReport__ol li {
    padding-left: 5px;
}

@media screen and (min-width: 1000px) {
    .ProgressReport__section h3 {
        font-size: 28px;
    }
}
