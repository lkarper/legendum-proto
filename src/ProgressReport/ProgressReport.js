import React, { useContext } from 'react';
import UserContext from '../contexts/UserContext';
import ChapterProgress from '../ChapterProgress/ChapterProgress';
import './ProgressReport.css';

const ProgressReport = (props) => {

    const context = useContext(UserContext);
    const { exercises } = context;

    return (
        <section className='ProgressReport__section'>
            <h3>User Progress</h3>
            <ol className='ProgressReport__ol'>
                {exercises
                    .sort((a, b) => a.id - b.id)
                    .map(exercise => <ChapterProgress key={exercise.id} exercise={exercise} />)
                }
            </ol>
        </section>
    )
}

export default ProgressReport;