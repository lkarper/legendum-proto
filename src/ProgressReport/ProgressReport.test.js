import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ProgressReport from './ProgressReport';

describe('ProgressReport component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ProgressReport />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<ProgressReport />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
