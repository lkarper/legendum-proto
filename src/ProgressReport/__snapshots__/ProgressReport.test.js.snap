// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressReport component renders the UI as expected 1`] = `
<ProgressReport>
  <section
    className="ProgressReport__section"
  >
    <h3>
      User Progress
    </h3>
    <ol
      className="ProgressReport__ol"
    />
  </section>
</ProgressReport>
`;
